server/core/region_tree.go

```diff
@@ -129,3 +129,24 @@ func (t *regionTree) scanRange(startKey []byte, f func(*metapb.Region) bool) {
 		return f(item.(*regionItem).region)
 	})
 }
+
+func (t *regionTree) getAdjacentRegions(region *metapb.Region) (*regionItem, *regionItem) {
+	item := &regionItem{region: &metapb.Region{StartKey: region.StartKey}}
+	var prev, next *regionItem
+	t.tree.DescendLessOrEqual(item, func(i btree.Item) bool {
+		if bytes.Compare(item.region.StartKey, i.(*regionItem).region.StartKey) == 0 {
+			return true
+		}
+		next = i.(*regionItem)
+		return false
+	})
+	item = &regionItem{region: &metapb.Region{StartKey: region.StartKey}}
+	t.tree.AscendGreaterOrEqual(item, func(i btree.Item) bool {
```

Comment:
The AscendGreaterOrEqual will iterate in [item, last], so I am confused that why you set prev here?

Reply:
the item store in region_tree is in reversed order, namely startkey "a" is greater than "b". https://github.com/Connor1996/pd/blob/e81d31ff04bdb282b66521f09f1562088e8869e7/server/core/region_tree.go#L30

Reply:
please add a comment